python/cudf/cudf/tests/test_json.py

```diff
@@ -1006,3 +1006,95 @@ def test_json_round_trip_gzip():
         fo.seek(loc)
         written_df = cudf.read_json(fo, orient="records", lines=True)
     assert_eq(written_df, df)
+
+
+@pytest.mark.parametrize(
+    "jsonl_string",
+    [
```

Comment:
What happens when mixed types include both structs and lists? Example:
`{"a":[123, [{"0": 123}, {}], {"1": 321}], "b":1.0}`

Reply:
It's an error. It will throw an exception.